shared/middleware/helpers/bucket_helper.rb

```diff
@@ -87,6 +123,21 @@ def create_or_replace(encrypted_channel_id, filename, body, version = nil, abuse
     response
   end
 
+  #
+  # Copy an object within a channel, creating a new object in the channel.
+  #
+  # @param [String] encrypted_channel_id - App-identifying token
+  # @param [String] filename - Destination name for new object
+  # @param [String] source_filename - Name of object to be copied
+  # @param [Hash] S3 response from copy operation
+  def copy(encrypted_channel_id, filename, source_filename)
+    owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
+
+    key = s3_path owner_id, channel_id, filename
+    copy_source = @bucket + '/' + s3_path(owner_id, channel_id, source_filename)
+    @s3.copy_object(bucket: @bucket, key: key, copy_source: copy_source)
```

Comment:
It might be worth catching [AWS::S3::Errors::NoSuchKey](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Errors/NoSuchKey.html) here and returning `{status: 'NOT_FOUND'}` as is done in the `get` method above, unless you think the client is generally likely to have fresh data when issuing a copy command.

Should animations be marked as abusive when an app is marked as abusive, as assets are? If so, there is a future work item here to specify `metadata` or `metadata_directive` on the S3 requests.